### ISSUE TITLE
Mem leak when keepalive is used

### ISSUE THREAD

**Issue by zdm** [node-fetch/node-fetch#1446]
node-fetch: 3.1.0
nodejs: 17.3.0

Memory working set is grows and not freed:
```js
import fetch from "node-fetch";
import http from "http";

const agent = new http.Agent( {
    "keepAlive": true,
    // keepAliveMsecs: 5000,
} );

var i = 0;

while ( 1 ) {

    const res = await fetch("http://example.com", { agent });
    await res.text();

    console.log( ++i, res.status );
}
```

For pure node http everything is ok:
```js
import http from "http";

const agent = new http.Agent( {
    "keepAlive": true,
    // keepAliveMsecs: 5000,
} );

var i = 0;

while ( 1 ) {
    const res = await new Promise( resolve => {
        http.get( "http://example.com/", { agent }, res => {
            res.on( "data", chunk => {} );

            res.on( "end", () => {
                resolve( { "status": res.statusCode } );
            } );
        } );
    } );

    console.log( ++i, res.status );
}
```

**Comment by dnalborczyk** [comment]
just bisected this. introduced in v3.0.0 (beta.10 works fine) with this PR: https://github.com/node-fetch/node-fetch/pull/1064

**Comment by tekwiz** [comment]
This is fixed in #1325.

**Comment by felipec** [comment]
Quite likely a duplicate of #1295 and fixed by #1474.

**Comment by olegade** [comment]
When will this be merged?

**Comment by jimmywarting** [comment]
waiting for @tekwiz response to https://github.com/node-fetch/node-fetch/pull/1325#pullrequestreview-853593710 - currently it seems as if we have two possible PRs with different approaches

**Comment by zdm** [comment]
It is possible to apply PR and close this?

**Comment by felipec** [comment]
Is this already fixed by #1474?
